### PR TITLE
refactor(console): update custom JWT default code and function type definition

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

Update custom JWT default code and function type definition.
For client credentials JWT customizer, the second parameter (`data`) of `getCustomJwtClaims` should always be undefined.

## Testing

Tested locally.

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

File: packages/console/src/pages/JwtClaims/utils/config.tsx [hunk 1]
```diff
@@ -55,10 +55,12 @@ declare global {
      * This function is called to get custom claims for the JWT token.
      * 
      * @param {${JwtCustomizerTypeDefinitionKey.ClientCredentialsPayload}} token -The JWT token.
+     * @param {undefined} data - Logto internal data placeholder that can be used to pass additional information (temporarily not used but should be kept for future use)
```
Comment (simeng-li) [inline packages/console/src/pages/JwtClaims/utils/config.tsx:58]:
```suggestion
     * @param {undefined} data - Logto internal data placeholder that can be used to pass additional information (undefined for M2M accesstoken)
```

File: packages/console/src/pages/JwtClaims/utils/config.tsx [hunk 2]
```diff
@@ -86,6 +88,7 @@ const defaultClientCredentialsJwtCustomizerCode = `/**
 * This function is called to get custom claims for the JWT token.
 *
 * @param {${JwtCustomizerTypeDefinitionKey.ClientCredentialsPayload}} token -The JWT token.
+* @param {undefined} data - Logto internal data placeholder that can be used to pass additional information (temporarily not used but should be kept for future use)
```
Comment (simeng-li) [inline packages/console/src/pages/JwtClaims/utils/config.tsx:91]: ditto

**Review by simeng-li — APPROVED** [review]
overall LGTM

**Review by charIeszhao — APPROVED** [review]

File: packages/console/src/pages/JwtClaims/utils/config.tsx [hunk 3]
```diff
@@ -55,10 +55,12 @@ declare global {
      * This function is called to get custom claims for the JWT token.
      * 
      * @param {${JwtCustomizerTypeDefinitionKey.ClientCredentialsPayload}} token -The JWT token.
+     * @param {undefined} data - Logto internal data placeholder that can be used to pass additional information (undefined for M2M accesstoken)
+     * @param {${JwtCustomizerTypeDefinitionKey.EnvironmentVariables}} envVariables - The environment variables.
      * 
      * @returns The custom claims.
      */
-    getCustomJwtClaims: (token: ${JwtCustomizerTypeDefinitionKey.ClientCredentialsPayload}, envVariables: ${JwtCustomizerTypeDefinitionKey.EnvironmentVariables}) => Promise<CustomJwtClaims>;
+    getCustomJwtClaims: (token: ${JwtCustomizerTypeDefinitionKey.ClientCredentialsPayload}, data: undefined, envVariables: ${JwtCustomizerTypeDefinitionKey.EnvironmentVariables}) => Promise<CustomJwtClaims>;
```
Comment (gao-sun) [inline packages/console/src/pages/JwtClaims/utils/config.tsx:63]: if it's always undefined, why would we need it anyway?

Comment (darcyYe) [reply packages/console/src/pages/JwtClaims/utils/config.tsx:63]: Need to align the function's interface definition under access token and client credentials use cases.

**Review by wangsijie — APPROVED** [review]

**Comment by darcyYe** [comment]
Discussed offline, update AZ function implementation and pass the use case (whether it is access token or client credentials) to avoid defining unused input parameters in `getCustomJwtClaims` function interface.

**Pull request closed without merging**